src/Resources/ResourceManager/SdkModels/Deployments/DeploymentOperationErrorInfo.cs

```diff
@@ -0,0 +1,142 @@
+                {
+                    sb.AppendLine().AppendFormat(ProjectResources.DeploymentOperationTargetInfoInErrror, operation.Properties.TargetResource.ResourceType,operation.Properties.TargetResource.ResourceName);                    
+                }
+                
+                ErrorMessages.Add(sb.ToString());
+            }            
+        }
+
+        public string GetErrorMessagesWithOperationId(string deploymentName = null)
+        {
+            if (ErrorMessages.Count == 0)
+                return String.Empty;
+
+            var sb = new StringBuilder();
+
+            int maxErrors = ErrorMessages.Count > MaxErrorsToShow
+               ? MaxErrorsToShow
+               : ErrorMessages.Count;
+
+            sb.AppendFormat(ProjectResources.DeploymentOperationOuterError, deploymentName, maxErrors, ErrorMessages.Count);
+            sb.Append(string.Join(Environment.NewLine, ErrorMessages.Take(maxErrors)));
+            sb.AppendLine().AppendFormat(ProjectResources.DeploymentOperationId, this.RequestId);
+            
+            return sb.ToString();
+        }
+
+
+        public void SetRequestIdFromResponseHeaders(HttpResponseMessage response)
+        {
+            RequestId = string.Empty;
+
+            if (response?.Headers != null && response.Headers.TryGetValues("x-ms-request-id", out IEnumerable<string> requestIdValues))
+            {
+                RequestId = string.Join(";", requestIdValues);
+            }
+        }
+        #endregion
+
+        #region Static Methods
+        public static string GetErrorMessageWithDetails(ErrorResponse error)
+        {
+            if (error == null) return null;
+
+            if (error.Details == null)
+            {
+                return string.Format(ProjectResources.DeploymentOperationErrorMessageNoDetails, error.Code, error.Message);
+            }
+
+            string errorDetail = null;
+
+            foreach (ErrorResponse detail in error.Details)
+            {
+                errorDetail += GetErrorMessageWithDetails(detail);
+            }
+
+            return string.Format(ProjectResources.DeploymentOperationErrorMessage, error.Code, error.Message, errorDetail);            
+        }
+
+        public static ErrorResponse DeserializeDeploymentOperationError(string statusMessage)
+        {
+            if (statusMessage == null)
+            {
+                return null;
+            }
+
+            ErrorResponse error = null;
+            dynamic dynamicStatusMessage;
+
+            try
+            {
+                dynamicStatusMessage = JsonConvert.DeserializeObject(statusMessage);
+                error = JsonConvert.DeserializeObject<ErrorResponse>(dynamicStatusMessage?.error?.ToString());
+            }
+            catch
+            {
+                // We'll ignore the exception if we can't properly deserialize the JSON.
```

Comment:
I'd recommend putting something into the error string for this. Maybe the exception message, and/or a message indicating deserialization failed and including the raw JSON so someone could figure out what happened. At minimum, I think it's better not to return null (unless null means something special to the caller).

Reply:
I might be wrong, but the assumption I made was if we can't deserialize an error returned from our service, then it's not clear to us and to the user what that error is. I could surface the exception message but I am not sure if the user could do anything about that exception. I might be overlooking something though - what do you think of this approach?

Reply:
Yes, that's true, but I think saying nothing is usually worse. Not only does the user not know what happened, but we will never find out it happened, either. If there is a common error mode that fails to deserialize, we could find and fix it if the errors are getting back to the user.

I think it's usually best to tell them everything you know. Maybe something like this:

Your deployment failed, but we were unable to get additional error information. The attempt to retrieve detailed information failed with this error:
Message: <message>
Stacktrace: <stacktrace>

Or something similar.

Reply:
Created a new PR to address some of the feedback. [https://github.com/Azure/azure-powershell/pull/11763](https://github.com/Azure/azure-powershell/pull/11763)

---
In reply to: [414766481](https://github.com/Azure/azure-powershell/pull/11644#discussion_r414766481) [](ancestors = 414766481)